resource/utilities/rq2.cpp

```diff
@@ -69,16 +225,137 @@ int info (resource_query_t &ctx)
     return rc;
 }
 
+static void process_args (json_t *options,
+                          int argc, char *argv[])
+{
+    int rc = 0;
+    int ch = 0;
+    std::string token;
+
+    while ((ch = getopt_long (argc, argv, OPTIONS, longopts, NULL)) != -1) {
+        switch (ch) {
+            case 'h': /* --help */
+                usage (0);
+                break;
+            case 'L': /* --load-file */
+                json_object_set_new(options, "load_file", json_string(optarg));
+                if (!fs::exists(optarg)) {
```

Comment:
Is there a reason why using `optarg` is a bad idea here? from the original function in resource-query, it grabbed the value from ctx after loading it. Using the json object, it is a fair bit more verbose to get the value however, an example can be seen on line 256.

Reply:
I'm a bit confused. Who said it was a bad idea? And it looks like you're using it anyway?

Reply:
No one said its a bad idea, I just want to check. In the original resource-query code, you would get something that looks like 
```
ctx->params.load_file = optarg;
                if (!fs::exists(ctx->params.load_file)) {
...
```
So it was explicitly getting the value from the ctx structure rather than from the variable optarg. I am just not sure why that decision was made and if there are any issues with using optarg instead?

Reply:
> I am just not sure why that decision was made

That's because the default parameters are set first and then overridden by the arguments if present. 
> if there are any issues with using optarg instead?

I don't see any issues with `optarg`.